pkg/bindinfo/binding_cache.go

```diff
@@ -92,7 +107,16 @@ func (fbc *fuzzyBindingCache) getFromMemory(sctx sessionctx.Context, fuzzyDigest
 	leastWildcards := len(tableNames) + 1
 	enableFuzzyBinding := sctx.GetSessionVars().EnableFuzzyBinding
 	for _, sqlDigest := range fbc.fuzzy2SQLDigests[fuzzyDigest] {
-		if bindings := bindingCache.GetBinding(sqlDigest); bindings != nil {
+		bindings := bindingCache.GetBinding(sqlDigest)
+		if intest.InTest && sctx.Value(GetBindingReturnNil) != nil {
```

Comment:
too many test-oriented variables? failpoint isn't work here?

wrap two if into one

Reply:
he logic here is a bit complex and requires two variables.